service/dataSourceService.go

```diff
@@ -269,32 +271,65 @@ func (service *DataSourceService) GetHarvestStatus(ctx context.Context, id strin
 func (service *DataSourceService) ConsumeReport(ctx context.Context, routingKey string, body []byte) []error {
 	var errors []error
 	if strings.Contains(routingKey, "harvested") {
-		var reports []model.HarvestReport
-		err := json.Unmarshal(body, &reports)
-		if err != nil {
-			errors = append(errors, err)
-		} else {
-			for _, report := range reports {
-				err := service.ReportsRepository.UpsertReports(ctx, report)
-				if err != nil {
-					errors = append(errors, err)
-				}
+		errors = service.consumeHarvestedReport(ctx, body)
+	} else if strings.Contains(routingKey, "reasoned") {
+		errors = service.consumeReasonedReport(ctx, body)
+	} else if strings.Contains(routingKey, "ingested") {
+		errors = service.consumeIngestedReport(ctx, routingKey, body)
+	}
+
+	return errors
```

Comment:
Trenger kanskje ikke lage liste for errors her.
```go
func (service *DataSourceService) ConsumeReport(ctx context.Context, routingKey string, body []byte) []error {
	switch {
	case strings.Contains(routingKey, "harvested"):
		return service.consumeHarvestedReport(ctx, body)
	case strings.Contains(routingKey, "reasoned"):
		return service.consumeReasonedReport(ctx, body)
	case strings.Contains(routingKey, "ingested"):
		return service.consumeIngestedReport(ctx, routingKey, body)
	default:
		return nil
	}
}
```

Reply:
trenger jeg da en nil-check når jeg iterer over lista med `range`?

Reply:
Tror du kan iterere `nil`, så lenge typen er en liste.

Reply:
```go
import "fmt"

func main() {
	for a := range foo() {
		fmt.Println(a)
	}
}

func foo() []bool {
	return nil
}
``

Reply:
dd96619